<launch>
<!--
  <node pkg="vicon"
        name="vicon"
        type="vicon"
        output="screen">
  </node>
-->

  <group ns="Coax56">
<!--
    <node pkg="vicon_calibrate"
          name="vicon_calibrate"
          type="vicon_calibrate"
          output="screen"
          clear_params="true">
      <remap from="~values" to="/vicon/values"/>
      <remap from="~names" to="/vicon/names"/>
      <param name="calibration" value="$(find coax_vsk)/calibration/Coax56.xml"/>
    </node>

    <node pkg="vicon2odometry"
          name="vicon2odometry"
          type="vicon2odometry"       
          output="screen">
      <remap from="~names" to="/vicon/names"/>
      <remap from="~values" to="/vicon/values"/>
      <remap from="~pose" to="/Coax56/vicon_calibrate/pose"/>
      <remap from="~odom" to="/Coax56/vicon2odometry/odom"/>
      <param name="vsk" value="$(find coax_vsk)/vsk/Coax56.vsk"/>

    </node>

    <node pkg="ipc_nav_msgs"
          name="nav_msgs_Odometry_publisher"
          type="nav_msgs_Odometry_publisher"
          output="screen">
      <remap from="~topic" to="/Coax56/vicon2odometry/odom"/>
      <param name="message" value="odom56"/>
    </node>
-->
    <node pkg="coax_server"
          name="coax"
          type="coax_server"
	  output="screen"
          args="/dev/tty.usbserial-A800f3Zv:1">
      <remap from="/coax_server/state" to="/Coax56/state"/>

      <remap from="/coax_server/control" to="/Coax56/control"/>
      <remap from="/coax_server/rawcontrol" to="/Coax56/rawcontrol"/>

      <remap from="/coax_server/reach_nav_state" to="/Coax56/reach_nav_state"/>
      <remap from="/coax_server/configure_comm" to="/Coax56/configure_comm"/>
      <remap from="/coax_server/set_timeout" to="/Coax56/set_timeout"/>
    </node>

    <node pkg="coax_interface"
          name="coax_interface"
          type="coax_interface"
	  output="screen">
      <remap from="/coax_interface/info" to="/Coax56/coax_info"/>
      <remap from="/coax_interface/rawcontrol" to="/Coax56/rawcontrol"/>
      <remap from="/coax_interface/control_mode" to="/Coax56/control_mode"/>

      <remap from="/coax_interface/state" to="/Coax56/state"/>
      <remap from="/coax_interface/trim" to="/Coax56/matlab_trim"/>
      <remap from="/coax_interface/nav_mode" to="/Coax56/matlab_nav_mode"/>
      <remap from="/coax_interface/raw_control" to="/Coax56/matlab_raw_control"/>

      <remap from="/coax_interface/reach_nav_state" to="/Coax56/reach_nav_state"/>
      <remap from="/coax_interface/configure_comm" to="/Coax56/configure_comm"/>
      <remap from="/coax_interface/set_timeout" to="/Coax56/set_timeout"/>

      <remap from="/coax_interface/set_control_mode" to="/Coax56/set_control_mode"/>

      <param name="frequency" value="100"/>
    </node>

    <node pkg="ipc_geometry_msgs"
          name="Quaternion_coax_info_publisher"
          type="geometry_msgs_Quaternion_publisher"
          output="screen">
      <remap from="~topic" to="/Coax56/coax_info"/>
      <param name="message" value="coax_info56"/>
    </node>

    <node pkg="ipc_geometry_msgs"
          name="Quaternion_trim_subscriber"
          type="geometry_msgs_Quaternion_subscriber"
          output="screen">
      <remap from="~topic" to="/Coax56/matlab_trim"/>
      <param name="message" value="trim56"/>
    </node>

    <node pkg="ipc_std_msgs"
          name="Bool_nav_mode_subscriber"
          type="std_msgs_Bool_subscriber"
          output="screen">
      <remap from="~topic" to="/Coax56/matlab_nav_mode"/>
      <param name="message" value="nav_mode56"/>
    </node>

    <node pkg="ipc_geometry_msgs"
          name="Quaternion_raw_control_subscriber"
          type="geometry_msgs_Quaternion_subscriber"
          output="screen">
      <remap from="~topic" to="/Coax56/matlab_raw_control"/>
      <param name="message" value="raw_control56"/>
    </node>

    <node pkg="ipc_geometry_msgs"
          name="Quaternion_control_mode_publisher"
          type="geometry_msgs_Quaternion_publisher"
          output="screen">
      <remap from="~topic" to="/Coax56/control_mode"/>
      <param name="message" value="control_mode56"/>
    </node>

  </group>


</launch>
